.github/actions/maven-cache/action.yml

```diff
@@ -61,4 +52,13 @@ runs:
           ~/.m2/repository/*/*
           !~/.m2/repository/org/keycloak
         key: ${{ steps.weekly-cache-key.outputs.key }}
-        enableCrossOsArchive: true
\ No newline at end of file
+        enableCrossOsArchive: true
+
+    # Downloadling Node.js often fails due to network issues, therefore we cache the artifacts downloaded by the frontend plugin.
+    - uses: actions/cache@v4
+      name: Cache Node.js related artifacts
+      with:
+        path: |
+          ~/.m2/repository/com/github/eirslett/node
+          ~/.m2/repository/com/github/eirslett/pnpm
+        key: ${{ runner.os }}-frontend-plugin-artifacts
```

Comment:
> The cache key is static. Therefore it will be cached once, and the cache is never refreshed for a different node or NPM version, which is then downloaded all the time. Consider adding a combined cache key for npm & node.

Done. The key for the cache now includes both the Node.js version and the PNPM version.

> This doesn't respect the setting `create-cache-if-it-doesnt-exist`. This could lead to a situation where there is no frontend being built, and then an empty cache is being created.

Like we discussed offline, the directory paths are very specific and would only create an artifact if the directories exist and contain files. Hence this should not be a problem for this specific cache action.